staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go

```diff
@@ -31,8 +31,12 @@ var (
 	ErrPatchContentNotMatchRetainKeys       = errors.New("patch content doesn't match retainKeys list")
 )
 
-func ErrNoMergeKey(m map[string]interface{}, k string) error {
-	return fmt.Errorf("map: %v does not contain declared merge key: %s", m, k)
+func ErrNoMergeKey(m1, m2 map[string]interface{}, k ...string) error {
+	if m2 == nil {
+		return fmt.Errorf("map: %v does not contain declared merge keys: %v", m1, k)
+	} else {
```

Comment:
You can drop the trailing `else` since there's a return in the first block of the conditional.